app/code/Magento/Customer/Controller/Address/Delete.php

```diff
@@ -18,10 +18,16 @@ public function execute()
         $addressId = $this->getRequest()->getParam('id', false);
 
         if ($addressId && $this->_formKeyValidator->validate($this->getRequest())) {
+            $customer = $this->customerRepository->getById($this->_getSession()->getCustomerId());
```

Comment:
@shandyDev there is a fatal error - "Call to a member function getById() on null" that can be reproduced following the next steps:
* Create customer on frontend
* Go to My Account->Address Book
* Add customer address.
* Go to My Account->Address Book
* Click Add New Address.
* Fill required fields and click Save Address
* Click Delete Address in Additional Address Entries block.

Reply:
Reproduced on test instance, soon will be a fix